SimTracker/SiPhase2Digitizer/plugins/PixelDigitizerAlgorithm.cc

```diff
@@ -240,3 +246,67 @@ bool PixelDigitizerAlgorithm::isAboveThreshold(const DigitizerUtility::SimHitInf
   } else
     return true;
 }
+//
+// -- Read Bad Channels from the Condidion DB and kill channels/module accordingly
+//
+void PixelDigitizerAlgorithm::module_killing_DB(const Phase2TrackerGeomDetUnit* pixdet) {
+  bool isbad = false;
+  uint32_t detID = pixdet->geographicalId().rawId();
+  int ncol = pixdet->specificTopology().ncolumns();
+  if (ncol < 0)
+    return;
+  std::vector<SiPixelQuality::disabledModuleType> disabledModules = siPixelBadModule_->getBadComponentList();
+
+  SiPixelQuality::disabledModuleType badmodule;
+  for (size_t id = 0; id < disabledModules.size(); id++) {
+    if (detID == disabledModules[id].DetID) {
+      isbad = true;
+      badmodule = disabledModules[id];
+      break;
+    }
+  }
```

Comment:
```suggestion
  for (const auto& mod : disabledModules) {
    if (detID == mod.DetID) {
      isbad = true;
      badmodule = mod;
      break;
    }
  }
```
this could be a range-based loop.